### PR TITLE
[WIP] [google_maps_flutter] Circle drawing support on google maps

### PR DESCRIPTION
**Pull request by awazgyawali**
This PR contains the following fixes:
1) Callback on tap event on the map with LatLng 
2) Ability to draw a circle on the map.

Note that this PR contains commits from my another PR #1049 which is currently active and contains the following changes 
1) Polyline support for Android and IOS
2) marker drag listener

I think it would be better to look/review this PR after the first one is accepted.

### CODE REVIEW

**Comment by isaacfi** [comment]
Hi @awazgyawali,

Any news about it? Did you commit with some other account? Please tell to Google team what appended? I believe that more than one are interested of this feature

**Comment by awazgyawali** [comment]
@isaacfi This PR depends upon #1049 thus after that PR is accepted, I will rebase it with the master.

**Comment by madtocc** [comment]
> #1049
@awazgyawali, any updates? #1049 was merged a few days ago. Thanks

**Comment by awazgyawali** [comment]
I am on a vacation, will refactor this PR once I get back. Probably coming the weekend.

**Comment by isaacfi** [comment]
@awazgyawali do you know who have to fix the conflicts? Is it an activity for Google team?
Regards

**Comment by awazgyawali** [comment]
I think this feature has been added already. Thus closing it.

File: packages/google_maps_flutter/android/src/main/java/io/flutter/plugins/googlemaps/CircleBuilder.java [hunk 1]
```diff
@@ -0,0 +1,67 @@
...
+  }
+
+  @Override
+  public void setConsumeTapEvents(boolean consumeTapEvents) {
+    circleOptions.clickable(consumeTapEvents);
+  }
+
+  @Override
+  public void setCenter(LatLng center) {
+    circleOptions.center(center);
```
Comment (anandgp) [inline packages/google_maps_flutter/android/src/main/java/io/flutter/plugins/googlemaps/CircleBuilder.java]: oomb

**Pull request closed without merging**